### PR TITLE
Update nrfxlib to 2.4.2

### PR DESCRIPTION
**Pull request by tdittr**
- Update nrfxlib
- Remove unnecessary FFI
- Adapt IPC handling to updated nrfxlib

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,19 +13,26 @@ keywords = ["nRF9160", "LTE", "GPS", "NB-IoT", "embedded"]
...
+] }
 nrf9160-pac = "0.12.2"
 arrayvec = { version = "0.7", default-features = false }
 at-commands = "0.5.2"
 no-std-net = "0.6.0"
 critical-section = "1.1"
 embassy-sync = "0.2.0"
 
+[patch.crates-io]
+nrfxlib-sys = { path = "../nrfxlib-sys" }
```
Comment (tdittr) [inline Cargo.toml:34]: This can be removed once `nrfxlib-sys` version 2.4.2 is released

**Comment by diondokter** [comment]
@tdittr Ok, I've let the CI run and it breaks on some formatting.
Also, is this supposed to be a draft still?

File: Cargo.toml (hunk 1 above)
Comment (diondokter) [reply Cargo.toml:34]: Oh, right that needs to happen before this gets merged. Gotcha

File: src/ffi.rs [hunk 2]
```diff
@@ -374,23 +333,45 @@ unsafe fn generic_free(ptr: *mut u8, heap: &crate::WrappedHeap) {
...
 /// Call this when we have an IPC IRQ. Not `extern C` as its not called by the
 /// library, only our interrupt handler code.
+// This function seems to be based on this verion in C:
+// https://github.com/NordicSemiconductor/nrfx/blob/98d6f433313a3d8dcf08dce25e744617b45aa913/drivers/src/nrfx_ipc.c#L146-L163
 pub unsafe fn nrf_ipc_irq_handler() {
     // Get the information about events that fired this interrupt
     let events_map = (*nrf9160_pac::IPC_NS::ptr()).intpend.read().bits();
 
+    #[cfg(feature = "defmt")]
+    defmt::trace!("IPC start");
```
Comment (diondokter) [inline src/ffi.rs:343]: We don't really want to log this. It was there for just debugging but is generally useless

File: src/ffi.rs [hunk 3]
```diff
@@ -374,23 +333,45 @@ unsafe fn generic_free(ptr: *mut u8, heap: &crate::WrappedHeap) {
...
+        }
     }
 
-    // Execute interrupt handler to provide information about events to app
-    let handler_addr = IPC_HANDLER.load(core::sync::atomic::Ordering::SeqCst);
-    let handler = core::mem::transmute::<usize, NrfxIpcHandler>(handler_addr);
-    let context = IPC_CONTEXT.load(core::sync::atomic::Ordering::SeqCst);
-    (handler)(events_map, context as *mut u8);
+    #[cfg(feature = "defmt")]
+    defmt::trace!("IPC done");
```
Comment (diondokter) [inline src/ffi.rs:374]: Same with this log

File: Cargo.toml (hunk 1 above)
Comment (tdittr) [reply Cargo.toml]: I changed it to a git dependency now, so CI can also find it, but we still need to wait until `nrfxlib-sys` pushed a new version.

**Comment by tdittr** [comment]
> Also, is this supposed to be a draft still?

Yes I wanted to keep it a draft until we can remove the `[patch.crates-io]`.

**Comment by diondokter** [comment]
Cool, thank you!

**Pull request merged**